### PR TITLE
feat(log): Log improvements for `ghost update` 

### LINKED ISSUE #562: Show help output when `ghost update` fails
If i run `ghost update` and the database migrations fail, i am not getting a hint to run

`ghost update --rollback`

It would be great to show a little hint on the bottom.

CLI: master
Ghost: latest

### PR DESCRIPTION
**Pull request by kirrg001**
closes #562

- be able to pass a command suggestion
- added basic toString fn to error module

### CODE REVIEW

File: lib/errors.js [hunk 1]
```diff
@@ -22,9 +22,11 @@ class CliError extends Error {
         // see https://nodejs.org/api/errors.html#errors_error_capturestacktrace_targetobject_constructoropt
         Error.captureStackTrace(this, this.constructor);
 
-        this.context = options.context || {};
         this.options = options;
         this.help = 'Please refer to https://docs.ghost.org/v1/docs/troubleshooting#section-cli-errors for troubleshooting.'
+
+        // additional error context
+        this.options.context = this.options.context || {};
```
Comment (kirrg001) [inline lib/errors.js]: When i print `this.context` it's never the context object which is passed in the options. It's always a huge object - not sure where this is coming from.

We also use `this.options.help` [here](https://github.com/TryGhost/Ghost-CLI/blob/1.4.0/lib/errors.js#L59)

File: lib/errors.js [hunk 2]
```diff
@@ -51,14 +53,18 @@ class CliError extends Error {
...
-
         if (this.options.help) {
             output += `${chalk.gray('Help:')} ${this.options.help}\n`;
         }
 
+        if (this.options.context.suggestion) {
+            output += `${chalk.green('Suggestion:')} ${this.options.context.suggestion}\n`;
+        }
+
+        if (verbose) {
```
Comment (kirrg001) [inline lib/errors.js]: reordered. it looks very unreadable if we first print the stack and then a small line with `Help:`.

**Comment by kirrg001** [comment]
gergregre

**Comment by kirrg001** [comment]
I can add tests if it get's approved 🙃

**Comment by AileenCGN** [comment]
@kirrg001 This PR seems like a good enhancement. Any objections, @acburdine?
Can you add some tests for it?

**Comment by kirrg001** [comment]
> I can add tests if it get's approved 🙃

🙃

**Comment by kirrg001** [comment]
@acburdine Could you pls review? Thanks 👍

File: lib/errors.js [hunk 3]
```diff
@@ -51,14 +53,18 @@ class CliError extends Error {
...
 
-        if (verbose) {
-            output += `${chalk.yellow('Stack:')} ${this.stack}\n`;
-        }
-
         if (this.options.help) {
             output += `${chalk.gray('Help:')} ${this.options.help}\n`;
         }
 
+        if (this.options.context.suggestion) {
```
Comment (acburdine) [inline lib/errors.js]: should just make this `this.options.suggestion` and remove the context stuff from above.

File: lib/errors.js [hunk 4]
```diff
@@ -101,8 +114,15 @@ class CliError extends Error {
...
+    constructor(options) {
+        options = options || {};
+
+        options.message = options.message || `Error occurred running command: '${options.cmd}'`;
+        super(options);
+    }
+
     toString(verbose) {
-        let output = chalk.red(`Error occurred running command: '${this.options.cmd}'\n\n`);
+        let output = super.toStringBasics();
```
Comment (acburdine) [inline lib/errors.js]: can just use `this.toStringBasics` here b/c the method is inherited.

Comment (kirrg001) [reply lib/errors.js]: yeah true 👍

**Comment by kirrg001** [comment]
updated

**Pull request merged**